plugins/inputs/elasticsearch_query/elasticsearch_query.go

```diff
@@ -0,0 +1,277 @@
+  ## Normally should be set to same as collection interval
+  query_period = "1m"
+
+  ## Optional parameters:
+  ## Lucene query to filter results
+  filter_query = "*"
+  ## Fields to aggregate values (must be numeric fields)
+  metric_fields = ["metric"]
+  ## Aggregation function to use on the metric fields
+  ## Valid values are: avg, sum, min, max, sum
+  metric_function = "avg"
+  ## Fields to be used as tags
+  ## Must be non-analyzed fields, aggregations are performed per tag
+  tags = ["field.keyword", "field2.keyword"]
+  ## Set to true to not ignore documents when the tag(s) above are missing
+  include_missing_tag = true
+  ## String value of the tag when the tag does not exist
+  ## Used when include_missing_tag is true
+  missing_tag_value = "null"
+`
+
+type ElasticsearchQuery struct {
+	URLs                []string `toml:"urls"`
+	Username            string
+	Password            string
+	EnableSniffer       bool
+	Tracelog            bool
+	Timeout             internal.Duration
+	HealthCheckInterval internal.Duration
+	Aggregations        []Aggregation `toml:"aggregation"`
+	tls.ClientConfig
+	httpclient *http.Client
+	ESClient   *elastic.Client
+	acc        telegraf.Accumulator
+}
+
+type Aggregation struct {
+	Index             string
+	MeasurementName   string
+	FilterQuery       string
+	QueryPeriod       internal.Duration
+	MetricFields      []string `toml:"metric_fields"`
+	DateField         string
+	Tags              []string `toml:"tags"`
+	IncludeMissingTag bool
+	MissingTagValue   string
+	MetricFunction    string
+}
+
+type aggKey struct {
+	measurement string
+	name        string
+	function    string
+	field       string
+}
+
+type aggregationQueryData struct {
+	aggKey
+	isParent    bool
+	aggregation elastic.Aggregation
+}
+
+func (e *ElasticsearchQuery) SampleConfig() string {
+	return sampleConfig
+}
+
+func (e *ElasticsearchQuery) Description() string {
+	return description
+}
+
+func (e *ElasticsearchQuery) init() error {
+	if e.URLs == nil {
+		return fmt.Errorf("Elasticsearch urls is not defined")
+	}
+
+	err := e.connectToES()
+	return err
+}
+
+func (e *ElasticsearchQuery) connectToES() error {
+
+	var clientOptions []elastic.ClientOptionFunc
+
+	if e.httpclient == nil {
+		httpclient, err := e.createHttpClient()
+		if err != nil {
+			return err
+		}
+		e.httpclient = httpclient
+	}
+
+	clientOptions = append(clientOptions,
+		elastic.SetHttpClient(e.httpclient),
+		elastic.SetSniff(e.EnableSniffer),
+		elastic.SetURL(e.URLs...),
+		elastic.SetHealthcheckInterval(e.HealthCheckInterval.Duration),
+	)
+
+	if e.Username != "" && e.Password != "" {
+		clientOptions = append(clientOptions,
+			elastic.SetBasicAuth(e.Username, e.Password),
+		)
+	}
+
+	if e.HealthCheckInterval.Duration == 0 {
+		clientOptions = append(clientOptions,
+			elastic.SetHealthcheck(false),
+		)
+	}
+
+	if e.Tracelog {
+		clientOptions = append(clientOptions,
+			elastic.SetTraceLog(log.New(os.Stdout, "", log.LstdFlags)),
+		)
+	}
```

Comment:
Where is this option set? Usually logging to stdout is an absolute exception. Can we remove this? Or at least guard with a config option even though I prefer the removal.

Reply:
That is for debugging the client, I agree it should not be in the release version as it pollutes the stdout and that's why I hid it from the config.

Reply:
I see. However, I'd like this removed from the code that goes to telegraf. If possible and necessary add some debug-logging using the telegraf logger...